woocommerce/woocommerce-ios
# [Mobile Payments] Cash on Delivery onboarding UI

Closes: #7475 


### Description

This adds the UI for a new onboarding step, which prompts the merchant to enable the Cash on Delivery payment gateway, if it's missing or disabled.

We want to encourage merchants to enable the `cod` gateway, because doing so allows them to take In-Person Payments for web orders.

This is the last onboarding step shown.

### Testing instructions


Using a store with COD disabled (check that it's disabled in `WP Admin > WooCommerce > Settings > Payments`)

1. Go to the hub menu
2. Tap Payments and wait for the spinner to complete
3. The onboarding incomplete footer message should appear: tap `Continue Setup`
4. The `Add Pay in Person to your checkout` onboarding screen should show

Using a store with COD enabled, repeat the above steps: the onboarding screen should not show.

### Notes
#### Buttons
The buttons on this screen do not do anything. The functionality will be added in future PRs.

#### Syncing
We do not currently do an extra refresh for Payment Gateways for this feature. This means the app will use the Payment Gateways synced at the last launch, as they are only refreshed [when the user launches the app or changes site](https://github.com/woocommerce/woocommerce-ios/blob/9bf2a29f09c4f1e0d4e28e9cf185d84cbb61be52/WooCommerce/Classes/Yosemite/DefaultStoresManager.swift#L471).

We could add a `synchronizePaymentGateways` action in `CardPresentPaymentsOnboardingUseCase.synchronizeStoreCountryAndPlugins`, but that would mean quite a lot of extra GET requests with little justification. I think we should (at most) do an extra sync of Payment Gateways after the user enables the COD gateway.

Happy to discuss this!

### Screenshots

![onboarding cod ui](https://user-images.githubusercontent.com/2472348/184910248-15088759-0fda-4427-b693-26009d795a50.jpg)



---
